eclipse-collections/src/main/java/org/eclipse/collections/impl/multimap/bag/SynchronizedBagMultimap.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) 2016 Shotaro Sano.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompany this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+package org.eclipse.collections.impl.multimap.bag;
+
+import java.io.Serializable;
+
+import net.jcip.annotations.GuardedBy;
+import org.eclipse.collections.api.bag.MutableBag;
+import org.eclipse.collections.api.block.function.Function;
+import org.eclipse.collections.api.block.function.Function2;
+import org.eclipse.collections.api.block.predicate.Predicate2;
+import org.eclipse.collections.api.multimap.Multimap;
+import org.eclipse.collections.api.multimap.bag.ImmutableBagMultimap;
+import org.eclipse.collections.api.multimap.bag.MutableBagMultimap;
+import org.eclipse.collections.api.tuple.Pair;
+import org.eclipse.collections.impl.factory.Multimaps;
+import org.eclipse.collections.impl.multimap.AbstractSynchronizedMultimap;
+
+public class SynchronizedBagMultimap<K, V>
+        extends AbstractSynchronizedMultimap<K, V>
+        implements MutableBagMultimap<K, V>, Serializable
+{
+    private static final long serialVersionUID = 1L;
+
+    public SynchronizedBagMultimap(MutableBagMultimap<K, V> newMultimap)
+    {
+        super(newMultimap);
+    }
+
+    public SynchronizedBagMultimap(MutableBagMultimap<K, V> newMultimap, Object newLock)
+    {
+        super(newMultimap, newLock);
+    }
+
+    /**
+     * This method will take a Multimap and wrap it directly in a SynchronizedMutableMultimap.
+     */
+    public static <K, V, M extends Multimap<K, V>> SynchronizedBagMultimap<K, V> of(M multimap)
+    {
+        if (multimap == null)
+        {
+            throw new IllegalArgumentException("cannot create a SynchronizedMutableMultimap for null");
+        }
+
+        return new SynchronizedBagMultimap<K, V>(Multimaps.mutable.bag.withAll(multimap));
```

Comment:
You can use diamond operator instead of `new SynchronizedBagMultimap<K, V>`
This can be changed to `new SynchronizedBagMultimap<>`